diff --git a/crypto/ecdsa/ecdsatest.c b/crypto/ecdsa/ecdsatest.c
index 856d042..037ffb8 100644
--- a/crypto/ecdsa/ecdsatest.c
+++ b/crypto/ecdsa/ecdsatest.c
@@ -260,7 +260,7 @@ int x9_62_tests(BIO *out, ENGINE *e)
     if (!change_rand())
         goto x962_err;
 
-    if (!x9_62_test_internal(out, NID_X9_62_prime192v1,
+    if (x9_62_test_internal(out, NID_X9_62_prime192v1,
                              "3342403536405981729393488334694600415596881826869351677613",
                              "5735822328888155254683894997897571951568553642892029982342",
                              e))
@@ -337,8 +337,8 @@ int test_builtin(BIO *out)
         if (nid == NID_ipsec4)
             continue;
 
-        /* Only testing curves supported by wolfEngine */
-        if (nid != NID_X9_62_prime192v1 && nid != NID_X9_62_prime256v1)
+        /* Only testing curves supported by wolfEngine and wolfCrypt FIPS */
+        if (nid != NID_X9_62_prime256v1)
             continue;
 
         /* create new ecdsa key (== EC_KEY) */
