diff --git a/apps/openssl.c b/apps/openssl.c
index a872e2c5..6fe79d59 100644
--- a/apps/openssl.c
+++ b/apps/openssl.c
@@ -127,6 +127,20 @@ int main(int argc, char *argv[])
     const char *prompt;
     ARGS arg;
     int first, n, i, ret = 0;
+    ENGINE *e;
+
+    ENGINE_load_dynamic();
+    e = ENGINE_by_id("libwolfengine");
+    if (e == NULL) {
+        printf("Failed to find wolfEngine.\n");
+        return 1;
+    }
+    if (ENGINE_ctrl_cmd(e, "enable_debug", 1, NULL, NULL, 0) != 1) {
+        printf("Failed to enable wolfEngine debug logging.\n");
+        return 1;
+    }
+    ENGINE_set_default(e, ENGINE_METHOD_ALL);
+
 
     arg.argv = NULL;
     arg.size = 0;
@@ -260,12 +274,14 @@ int main(int argc, char *argv[])
 
     BIO_free(bio_in);
     BIO_free_all(bio_out);
+    ENGINE_finish(e);
     apps_shutdown();
 #ifndef OPENSSL_NO_CRYPTO_MDEBUG
     if (CRYPTO_mem_leaks(bio_err) <= 0)
         ret = 1;
 #endif
     BIO_free(bio_err);
+    ENGINE_cleanup();
     EXIT(ret);
 }
 
